iroh-sync/src/store/fs.rs

```diff
@@ -237,6 +250,95 @@ impl super::Store for Store {
     fn content_hashes(&self) -> Result<Self::ContentHashesIter<'_>> {
         ContentHashesIterator::create(&self.db)
     }
+
+    fn register_useful_peer(&self, namespace: NamespaceId, peer: crate::PeerIdBytes) -> Result<()> {
+        let peer = &peer;
+        let namespace = namespace.as_bytes();
+        // calculate nanos since UNIX_EPOCH for a time measurement
+        let nanos = std::time::UNIX_EPOCH
+            .elapsed()
+            .map(|duration| duration.as_nanos() as u64)?;
+        let write_tx = self.db.begin_write()?;
+        {
+            let mut peers_table = write_tx.open_multimap_table(NAMESPACE_PEERS_TABLE)?;
+            let mut namespace_peers = peers_table.get(namespace)?;
+
+            // get the oldest entry since it's candidate for removal
+            let maybe_oldest = namespace_peers.next().transpose()?.map(|guard| {
+                let (oldest_nanos, &oldest_peer) = guard.value();
+                (oldest_nanos, oldest_peer)
+            });
+            match maybe_oldest {
+                None => {
+                    // the table is empty so the peer can be inserted without further checks since
+                    // super::PEERS_PER_DOC_CACHE_SIZE is non zero
+                    drop(namespace_peers);
+                    peers_table.insert(namespace, (nanos, peer))?;
+                }
+                Some((oldest_nanos, oldest_peer)) => {
+                    let oldest_peer = &oldest_peer;
+
+                    if oldest_peer == peer {
+                        // oldest peer is the current one, so replacing the entry for the peer will
+                        // maintain the size
+                        drop(namespace_peers);
+                        peers_table.remove(namespace, (oldest_nanos, oldest_peer))?;
+                        peers_table.insert(namespace, (nanos, peer))?;
+                    } else {
+                        // calculate the len in the same loop since calling `len` is another fallible operation
+                        let mut len = 1;
```

Comment:
why does this star at 1?

Reply:
Because sigue it's some the iterator already advanced by one

Reply:
ah yes 😅